module/VuFindApi/src/VuFindApi/Controller/SearchApiController.php

```diff
@@ -304,89 +336,161 @@ public function searchAction()
         $request = $this->getRequest()->getQuery()->toArray()
             + $this->getRequest()->getPost()->toArray();
 
+        $isCursorSearch = ($request['resumptionToken'] ?? false);
+        try {
+            if ($isCursorSearch) {
+                $response = $this->doCursorSearch($request);
+            } else {
+                $response = $this->doDefaultSearch($request);
+            }
+        } catch (Exception $e) {
+            return $this->output([], self::STATUS_ERROR, 400, $e->getMessage());
+        }
+        return $this->output($response, self::STATUS_OK);
+    }
+
+    /**
+     * Perform a search using page in Solr
+     *
+     * @param array $request Array containing combination of post and get request params
+     *
+     * @return array Response to be sent for the user
+     *               - records: Records found
+     *               - resultCount: Total result count
+     *               - facets: array containing facets for the result
+     */
+    protected function doDefaultSearch(array $request): array
+    {
         if (
-            isset($request['limit'])
-            && (!ctype_digit($request['limit'])
+            isset($request['limit']) && (!ctype_digit($request['limit'])
```

Comment:
The indentation changes here make this harder for me to read -- I think it might be good to keep `isset($request['limit'])` on its own line, and move the `&& (!ctype_digit($request['limit'])` down to the next line so that all of the OR clauses are together. By breaking up the nested OR parentheses into multiple lines, this implies an association that isn't correct, and it confused me. If doing this would make the second line too long, let me know if you'd like to brainstorm other options... one possibility might be to define `$limit` before the validation, since that would let us simplify all the `$request` references in the check, and I don't see that it would break the logic in any way.